### PR TITLE
bpo-46891: Fix creating a new instance of a module subclass with slots

### PR DESCRIPTION
**Pull request by markshannon**
https://bugs.python.org/issue46891

### CODE REVIEW

File: Lib/test/test_module.py [hunk 1]
```diff
@@ -346,6 +346,25 @@ def test_repeated_attribute_pops(self):
...
     # frozen and namespace module reprs are tested in importlib.
 
+    def test_subclass_with_slots(self):
+        # In 3.11alpha this crashed, as the slots weren't NULLed.
+
+        class ModuleWithSlots(ModuleType):
+            __slots__ = ("a", "b")
+
+            def __init__(self, name):
+                super().__init__(name)
```
Comment (vstinner) [inline Lib/test/test_module.py:356]: What's the purpose of this method? Maybe assign a and/or b attribute if you want to keep it? Or remove it?

**Review by vstinner — APPROVED** [review]
LGTM, I just have a minor question/remark on the test.

I'm happy to see the _PyType_AllocNoTrack() function that I added being used :-) Objects should only be tracked once their visit function can be safely called.

File: Lib/test/test_module.py (hunk 1 above)
Comment (ronaldoussoren) [reply Lib/test/test_module.py:356]: The important bit is having ``__slots__`` in the class definition, I just checked and leaving out the ``__init__`` method still reproduces the crash that this PR fixes.

Comment (vstinner) [reply Lib/test/test_module.py:356]: Well, I approved the PR. @markshannon can decide to keep or remove the method, it doesn't really matter ;-)

**Pull request merged**